Sources/Filters/Core/ThresholdPoints/index.js

```diff
@@ -0,0 +1,251 @@
+import macro from 'vtk.js/Sources/macros';
+import vtkCellArray from 'vtk.js/Sources/Common/Core/CellArray';
+import vtkDataArray from 'vtk.js/Sources/Common/Core/DataArray';
+import vtkPoints from 'vtk.js/Sources/Common/Core/Points';
+import vtkPolyData from 'vtk.js/Sources/Common/DataModel/PolyData';
+import { POLYDATA_FIELDS } from 'vtk.js/Sources/Common/DataModel/PolyData/Constants';
+
+const { vtkErrorMacro } = macro;
+
+const OperationType = {
+  Below: 'Below',
+  Above: 'Above',
+};
+
+// Function to perform binary search on a sorted array
+function binarySearch(items, value) {
+  let firstIndex = 0;
+  let lastIndex = items.length - 1;
+  let middleIndex = Math.floor((lastIndex + firstIndex) / 2);
+
+  while (items[middleIndex] !== value && firstIndex < lastIndex) {
+    if (value < items[middleIndex]) {
+      lastIndex = middleIndex - 1;
+    } else if (value > items[middleIndex]) {
+      firstIndex = middleIndex + 1;
+    }
+    middleIndex = Math.floor((lastIndex + firstIndex) / 2);
+  }
+
+  return {
+    found: items[middleIndex] === value,
+    index: Math.max(
+      items[middleIndex] < value ? middleIndex + 1 : middleIndex,
+      0
+    ),
+  };
+}
+
+function camelize(str) {
+  return str
+    .replace(/(?:^\w|[A-Z]|\b\w)/g, (letter) => letter.toUpperCase())
+    .replace(/\s+/g, '');
+}
+// ----------------------------------------------------------------------------
+// vtkThresholdPoints methods
+// ----------------------------------------------------------------------------
+
+function vtkThresholdPoints(publicAPI, model) {
+  // Set our classname
+  model.classHierarchy.push('vtkThresholdPoints');
+
+  publicAPI.requestData = (inData, outData) => {
+    const input = inData[0];
+    const output = vtkPolyData.newInstance();
+    outData[0] = output;
+
+    if (model.criterias.length === 0) {
+      output.shallowCopy(input);
```

Comment:
@floryst @jourdain I was surprised to have to shallow copy. 
If I simply do outputData[0] = input, then the shader does not refresh.
This happens when the threshold filter has previously generated a new ouput polydata and now it simply returns an old polydata. There must be some kind of time stamp issue. I'm not sure if that should be fixed instead in the mapper. WDYT ?

Reply:
It's probably b/c `input.getMTime()` is older than `output.getMTime()` at that point, so there's probably a `shouldUpdate` flag set to false in the vtk algorithm macro. `outputData[0] = input` would probably work if `input.modified()` was called as well, but that's probably undesirable, so shallowCopy seems fine to me.